plexupdate-dsm7.sh

```diff
@@ -0,0 +1,34 @@
+#!/bin/bash
+
+# Script to automagically update Plex Media Server on Synology NAS
+#
+# Must be run as root.
+#
+# @author @martinorob https://github.com/martinorob
+# https://github.com/martinorob/plexupdate/
+
+#!/bin/bash
+mkdir -p /tmp/plex/ > /dev/null 2>&1
+token=$(cat /volume1/Plex/Library/Application\ Support/Plex\ Media\ Server/Preferences.xml | grep -oP 'PlexOnlineToken="\K[^"]+')
```

Comment:
Should be "/var/packages/PlexMediaServer/home/Plex\ Media\ Server/Preferences.xml" since the Plex folder is toast now, right?

Reply:
Thanks, I didn't realise there was a new location for package.

I had issues upgrading Plex after installing DSM 7, and based on posts by a Plex team member on their forums just updated permissions of the `Plex` shared folder to give ownership to the new `PlexMediaServer` system user. At that point the new Plex package was able to read library data and was working correctly.

Currently `/var/packages/PlexMediaServer/home/Plex\ Media\ Server/Preferences.xml` looks like its missing the `PlexOnlineToken` and just generally in a default state (missing other values). Does Plex need to be told to use this new directory for storing its data? Do I need to migrate the data from the existing `Plex` shared folder?

Reply:
My understanding based on the forums is the DSM installer migrates the "Plex" Share to the new system on install and then you can just delete the "Plex" share (and that's that I've gone ahead and done). Took me a while to dig up the new preferences location but the "Preferences.xml" has my token. `cat /var/packages/PlexMediaServer/home/Plex\ Media\ Server/Preferences.xml | grep -oP 'PlexOnlineToken="\K[^"]+'` also returns my token.

Reply:
Got it. I didn't realise the migration failure I experienced was a permissions error at first, so I manually uninstalled Plex and the installed the DSM 7 package, bypassing the migration.

Going to try and fix it, in the meantime I've updated the path. Thanks for bringing it to my attention!

Reply:
FYI, I've been doing some more exploring with the new Plex setup. "/volume1/@apphome/PlexMediaServer/Plex\ Media\ Server/Preferences.xml" seems to be an equally valid path. Not sure which one is better to use, but thought it worth noting.

Reply:
Mine was found at "/volume1/PlexMediaServer/AppData/Plex\ Media\ Server/Preferences.xml"

Reply:
> Mine was found at "/volume1/PlexMediaServer/AppData/Plex\ Media\ Server/Preferences.xml"

Yeah, they keep moving it around, though I think it's "set" now. Your path is the current working path for me as well.

Reply:
@martinorob any issues with merging this?

Reply:
@martinorob 

Thank you for this excellent script.

My path was the same as @Fifteen15Studios's, but the `${VOLUME}` and `@apphome` variables in his suggestion didn't work for me on Synology DSM 7.0.1 and PlexMediaServer 1.25.2.5319. 

My token variable ended up looking like this:
`token=$(cat /volume1/PlexMediaServer/AppData/Plex\ Media\ Server/Preferences.xml | grep -oP 'PlexOnlineToken="\K[^"]+')`

To take it one step further, it appears that the `newversion` and `curversion` variables return the version number in slightly different formats, resulting in an attempted upgrade every time it runs.

Here's what it returns as of 2021-12-20
> New Ver: 1.25.2.5319-c43dc0277
> Cur Ver: 1.25.2.5319-7000

I appended some `sed` to the end of those variables...
`newversion=$(echo $jq | jq -r '.nas."Synology (DSM 7)".version'|sed 's/-.*//')`
`curversion=$(synopkg version "PlexMediaServer"|sed 's/-.*//')`

And now it returns:
> New Ver: 1.25.2.5319
> Cur Ver: 1.25.2.5319

I hope this helps. I'm still learning GitHub, so I'm not sure how to propose these changes otherwise.

Reply:
Thanks @OverengineeredNetwork! I was (finally!) tired of Plex unnecessarily updating so I ended up just piping it through the `cut` command (I like `sed` just fine but for this `cut` should be plenty):
`newversion=$(echo $jq | jq -r '.nas."Synology (DSM 7)".version' | cut -d'-' -f1)`

Reply:
I also checked to see if I needed to do this for the `curversion` variable as well (I'm on a "-<number>" version of Plex currently) and it works just fine with the `dpkg --compare-versions...` command. So you shouldn't need it on the `curversion` line.

Reply:
Neat @starryknight64 👍 

It sounds like you did what @Fifteen15Studios did in his script: https://github.com/martinorob/plexupdate/pull/8#issuecomment-997974182

Full disclosure: I only used `sed` because it was the first (workable) solution I could find on stackexchange for truncating a string 😂. Then I found out that @Fifteen15Studios was way ahead of me. 

I’m glad you were able to adapt it to your needs. Check out what @Fifteen15Studios did in his script. I think he made some other handy changes too.

Reply:
Yeah! I noticed that not long after my comment. Thanks! :)

Reply:
hmm. I've swapped out the curversion and newversion lines with those mentioned above but am still getting 

New Ver: 1.25.2.5319-c43dc0277
Cur Ver: 1.25.2.5319-7000

What should the latest and best script look like?

Reply:
@divadiow I would just go with @Fifteen15Studios’s script, as it seems to have many improvements not implemented in this pull request (see https://github.com/martinorob/plexupdate/pull/8#issuecomment-997974182). 

If you implemented the `sed` or `cut` commands we discussed here and it still doesn’t work, I would also want to make sure the script you edited is the one that is being run. 

Also, you might consider running the commands manually, line by line, to see why it’s not transforming the version numbers correctly.

Reply:
thanks @OverengineeredNetwork well for a start I was running the wrong damn script. I guess I didn't change the task location 🙄 

anyway. It's running now with the seds but I was getting no result output in the script text. I changed the location of the resultfile and added a cat at the end and now my output to file and the subsequent email alert contains "no new version" as expected.

working script is this 

```
#!/bin/bash

# Script to automagically update Plex Media Server on Synology NAS
#
# Must be run as root.
#
# @author @martinorob https://github.com/martinorob
# https://github.com/martinorob/plexupdate/

# Volume where Plex is installed
VOLUME=$(echo "/volume1")
# Location of temp folder for this script
tmpFolder="/volume1/Scripts"
resultFile=$tmpFolder/plexupdateresult.txt

# Get information we need
token=$(cat ${VOLUME}/PlexMediaServer/AppData/Plex\ Media\ Server/Preferences.xml | grep -oP 'PlexOnlineToken="\K[^"]+')
url=$(echo "https://plex.tv/api/downloads/5.json?channel=plexpass&X-Plex-Token=$token")
jq=$(curl -s ${url})

# Get version numbers
# For some reason the part after the dash changes, so exclude it
newversion=$(echo $jq | jq -r '.nas."Synology (DSM 7)".version'|sed 's/-.*//')
echo "New Ver: $newversion" > $resultFile
curversion=$(synopkg version "PlexMediaServer"|sed 's/-.*//')
echo "Cur Ver: $curversion" >> $resultFile

# Compare version numbers
if [ "$newversion" != "$curversion" ]
# New Version Available
then
	/usr/syno/bin/synonotify PKGHasUpgrade '{"%PKG_HAS_UPDATE%": "Plex Media Server"}'
	CPU=$(uname -m)
	url=$(echo "${jq}" | jq -r '.nas."Synology (DSM 7)".releases[] | select(.build=="linux-'"${CPU}"'") | .url')
	# Download the file
	/bin/wget $url -P $tmpFolder/
	# Get filename from URL
	slashes=$(awk -F"/" '{print NF-1}' <<< "${url}")
	filename=$(echo $url | cut -d '/' -f $(expr $slashes + 1))
	# Install the file
	/usr/syno/bin/synopkg install $tmpFolder/$filename >> $resultFile
	sleep 30
	/usr/syno/bin/synopkg start "Plex Media Server"
	rm -rf $tmpFolder/*.spk
else
	echo "No new version available" >> $resultFile
cat $resultFile
fi
exit
```

output has this

```
New Ver: 1.25.3.5409
Cur Ver: 1.25.3.5409
No new version available
```